### PR TITLE
Patch

### LINKED ISSUE #48: Patch Applicator
The patch applicator is responsible for taking a set of patches, and then executing the patches in the order provided.

Steps to executing a patch:
1. copy over patch files into `./patch` of the project directory
2. read in patch .yaml files
3. execute .yaml files (`maintenance` first if found, followed by `patch`)
4. clean up (remove `./patch`)

Due to the need to do the patches in order, this should be done synchronously. Maintenance patches should be applied before core patches, and the order of each patch yaml file is important as well and should be executed sequentially.

patch = {
  version: "0.21.0",
  patch: { ... },
  maintenance: { ... }
}
```

Patch yaml file actions are to be executed via the repo manipulator methods (edit, move, copy, rename, etc...).

### LINKED ISSUE #49: File Manipulator
Patch yaml files describe changes that are to be made to the Snap project repository. These changes may require modification to file contents, replacement of files, deletion of files or other file changes. The patch runner will call these various file manipulator functions as it executes the contents of the patch files.

Initial set of actions:
* edit

#### edit
Should support edit of YAML and JSON files via the following actions: 'update' and 'remove'
##### update
updates or adds a key value
##### remove
remove keys

### PR DESCRIPTION
**Pull request by dkonieczek**

### CODE REVIEW

**Comment by korgon** [comment]
@dkonieczek The `patch` command should be near complete now. We may need to do some final testing to verify. Additionally all commands have been manually tested after the great refactor. Please review the changes and test if time allows.

File: src/cli.js [hunk 1]
```diff
@@ -252,7 +292,36 @@ export async function cli(args) {
...
 		case 'whoami': {
-			console.log(`${chalk.blue(options.context.user.name)} (${chalk.green(options.context.user.login)})`);
+			console.log(`${chalk.blue(options.user.name)} (${chalk.green(options.user.login)})`);
+			break;
+		}
+
+		case 'patch': {
+			function showPatchHelp() {
+				help({ command: 'help', args: ['patch'] });
+				// TODO: add patch to help
```
Comment (dkonieczek) [inline src/cli.js]: Looks like this todo can be removed.

**Review by dkonieczek — COMMENTED** [review]
Code looks good. Tested all the commands. Here are some notes

snapfu secrets verify
- fails to run

Screen Shot 2022-08-16 at 11 43 52 AM

snapfu patch list
- do you think we should suppress the git pull output? 
- when the repo is up to date, it still logs "Updating snapfu-patches... Already up to date."

Screen Shot 2022-08-16 at 11 47 21 AM

Screen Shot 2022-08-16 at 11 49 15 AM

snapfu patch apply (without a version specified)
- Displays "Patch version invalid" - should this display the help message instead? (see above screenshot)

snapfu patch apply
- Consider changing text color of the patch file name, contrast not so great

Screen Shot 2022-08-16 at 11 51 26 AM


snapfu recs list/sync
- output of each line is very long and I'd assume most people use a smaller terminal window. Can we make these links clickable with a short url? Or maybe just have a single link at the end that points to the main templates in the SMC.

Screen Shot 2022-08-16 at 11 53 05 AM

**Review by korgon — APPROVED** [review]
Approved.  (stamp)

**Pull request merged**